pkg/controller/perconaxtradbcluster/perconaxtradbcluster_controller.go

```diff
@@ -235,6 +236,43 @@ func (r *ReconcilePerconaXtraDBCluster) deploy(cr *api.PerconaXtraDBCluster) err
 		return err
 	}
 
+	secretObj := corev1.Secret{}
+	err = r.client.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: nodeSet.Namespace,
+			Name:      cr.Spec.PXC.SSLSecretName,
+		},
+		&secretObj)
+	if len(secretObj.Name) == 0 && cr.Spec.PXC.AllowUnsafeConfig == false {
+		var (
+			issuerKind = "ClusterIssuer"
+			issuerName = cr.Namespace + cr.Spec.PXC.SSLSecretName + "-ca"
+		)
+		issuer := cm.ClusterIssuer{}
+		selfsignIssuer := cm.SelfSignedIssuer{}
+		issuer.Namespace = nodeSet.Namespace
+		issuer.Kind = issuerKind
+		issuer.Name = issuerName
+		issuer.Spec.SelfSigned = &selfsignIssuer
+		err = r.client.Create(context.TODO(), &issuer)
+		if err != nil {
+			log.Info("issuer: " + err.Error())
+		}
+		certificate := cm.Certificate{}
+		certificate.Namespace = nodeSet.Namespace
+		certificate.Kind = "Certificate"
+		certificate.Name = cr.Namespace + cr.Spec.PXC.SSLSecretName + ".com"
+		certificate.Spec.SecretName = cr.Spec.PXC.SSLSecretName
+		certificate.Spec.CommonName = cr.Spec.PXC.SSLSecretName + "-pxc"
+		certificate.Spec.IsCA = true
+		certificate.Spec.IssuerRef.Name = issuerName
+		certificate.Spec.IssuerRef.Kind = issuerKind
+		err = r.client.Create(context.TODO(), &certificate)
+		if err != nil {
+			log.Info("certificate: " + err.Error())
+		}
+	}
```

Comment:
I believe we have to move all this cert functionality into the separate method. Now it wedges into the code and brokes current "flow".  We start to create a nodeSet then 35 lines of certificated related code and then again back to the nodeSet creation. Take a look at the rest of this method.